x-pack/legacy/plugins/infra/public/pages/logs/analysis/page_setup_content.tsx

```diff
@@ -4,13 +4,212 @@
 import { useTrackPageview } from '../../../hooks/use_track_metric';
 
-export const AnalysisSetupContent = () => {
+interface AnalysisSetupContentProps {
+  setupMlModule: (startTime: number | undefined, endTime: number | undefined) => Promise<any>;
+  isSettingUp: boolean;
+}
+
+const startTimeLabel = i18n.translate('xpack.infra.analysisSetup.startTimeLabel', {
+  defaultMessage: 'Start time',
+});
+const endTimeLabel = i18n.translate('xpack.infra.analysisSetup.endTimeLabel', {
+  defaultMessage: 'End time',
+});
+const startTimeDefaultDescription = i18n.translate(
+  'xpack.infra.analysisSetup.startTimeDefaultDescription',
+  {
+    defaultMessage: 'now',
+  }
+);
+const endTimeDefaultDescription = i18n.translate(
+  'xpack.infra.analysisSetup.endTimeDefaultDescription',
+  {
+    defaultMessage: 'indefinitely',
+  }
+);
+const clearStartTimeLabel = i18n.translate('xpack.infra.analysisSetup.clearStartTimeLabel', {
+  defaultMessage: 'Clear start time',
+});
+const clearEndTimeLabel = i18n.translate('xpack.infra.analysisSetup.clearEndTimeLabel', {
+  defaultMessage: 'Clear end time',
+});
+
+function selectedDateToParam(selectedDate: Moment | null) {
+  if (selectedDate) {
+    return selectedDate.unix();
+  }
+  return undefined;
+}
+
+export const AnalysisSetupContent: React.FunctionComponent<AnalysisSetupContentProps> = ({
+  setupMlModule,
+  isSettingUp,
+}) => {
   useTrackPageview({ app: 'infra_logs', path: 'analysis_setup' });
   useTrackPageview({ app: 'infra_logs', path: 'analysis_setup', delay: 15000 });
 
-  return <div>Setup</div>;
+  const [startTime, setStartTime] = useState<Moment | null>(null);
+  const [endTime, setEndTime] = useState<Moment | null>(null);
+
+  const now = useMemo(() => moment(), []);
+  const selectedEndTimeIsToday = !endTime || endTime.isSame(now, 'day');
+
+  const onClickCreateJob = () =>
+    setupMlModule(selectedDateToParam(startTime), selectedDateToParam(endTime));
+
+  return (
+    <AnalysisSetupPage>
+      <EuiPageBody>
+        <AnalysisPageContent
+          verticalPosition="center"
+          horizontalPosition="center"
+          data-test-subj="analysisSetupContent"
+        >
+          <EuiPageContentHeader>
+            <EuiPageContentHeaderSection>
+              <EuiTitle size="m">
+                <h3>
+                  <FormattedMessage
+                    id="xpack.infra.analysisSetup.analysisSetupTitle"
+                    defaultMessage="Enable Machine Learning analysis"
+                  />
+                </h3>
+              </EuiTitle>
+            </EuiPageContentHeaderSection>
+          </EuiPageContentHeader>
+          <EuiPageContentBody>
+            <EuiText size="s">
+              <FormattedMessage
+                id="xpack.infra.analysisSetup.analysisSetupDescription"
+                defaultMessage="Use Machine Learning to automatically categorize log messages, and detect anomalous event counts and rare log messages."
```

Comment:
As we're not shipping this with the full feature set for 7.4, I'd change this to:

```suggestion
                defaultMessage="Use Machine Learning to automatically detect anomalous log rate counts"
```